### PR TITLE
Add static content intl(locale) support

### PR DESCRIPTION
**Pull request by awesomejerry**
related to #684, #686  

support:

```
src/content/about.en.jade
src/content/about/en.jade
```

### CODE REVIEW

**Comment by langpavel** [comment]
Hi @awesomejerry
This code is build on `feature/react-intl` which internally uses Redux, but you implement this through `context`. This is not preferred way of doing fetching in Redux branches. If you do this by redux, you will see that you can entirely drop `localeDetect` HoC which is also not good example of HoC..

You can look at `LanguageSwitcher` and try rewrite `Content` code to use Redux.

I know I should rewrite content fetching too and I probably do this soon but for now I will do not merge this.

Anyway Thanks! And ping me in week if nothing happens here :smiley:

**Comment by awesomejerry** [comment]
The main reason I wrote `localeDetect` is that I don't want to mess up with the original code as much as possible (which is not meant for HoC demonstration apparently), trying to make things happen just inside a new component (that's why the usage of internal state). Furthermore, the locale content is only used by `Component`, so I didn't use redux or even create a new reducer. But hey, making things consistent might be great though. I'll make some changes by the end of the week.

FYI, I'm gonna make `Component` a stateful component in order to detect client side language switch. Dispatch action whenever the locale changes and take the content from `state.content.data`. Please let me know if there's any recommended ways of doing so.

**Comment by awesomejerry** [comment]
Hi @langpavel , I've updated the PR to redux style. Please check it. Thanks!

File: src/actions/content.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    content(path:$path,locale:$locale) {
+      title, content,
+    }
+  }
+`;
+
+export function getLocaleContent(path, locale) {
+  return async (dispatch, getState, { graphqlRequest }) => {
+    dispatch({
+      type: GET_LOCALE_CONTENT,
```
Comment (langpavel) [inline src/actions/content.js]: * `FETCH_CONTENT_START`
* add more data to payload, path, locale
* you can make locale optional witch default value from `getState().intl.locale`

File: src/actions/content.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+const query = `
+  query($path:String!,$locale:String!) {
+    content(path:$path,locale:$locale) {
+      title, content,
+    }
+  }
+`;
+
+export function getLocaleContent(path, locale) {
```
Comment (langpavel) [inline src/actions/content.js]: use object for passing in arguments
```js
export function getLocaleContent({ path, locale })
```

File: src/constants/index.js [hunk 3]
```diff
@@ -4,3 +4,6 @@ export const SET_RUNTIME_VARIABLE = 'SET_RUNTIME_VARIABLE';
 export const SET_LOCALE_START = 'SET_LOCALE_START';
 export const SET_LOCALE_SUCCESS = 'SET_LOCALE_SUCCESS';
 export const SET_LOCALE_ERROR = 'SET_LOCALE_ERROR';
+export const GET_LOCALE_CONTENT = 'GET_LOCALE_CONTENT';
```
Comment (langpavel) [inline src/constants/index.js]: `FETCH_CONTENT_START` note it is very good practice to use suffixes consistently, see for example [react-redux-loading-bar](https://mironov.github.io/react-redux-loading-bar/)

File: src/reducers/content.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  },
+};
+
+export default function runtime(state = defaultState, action) {
+  switch (action.type) {
+    case GET_LOCALE_CONTENT:
+      return {
+        ...state,
+        isFetching: true,
+      };
```
Comment (langpavel) [inline src/reducers/content.js]:
```js
const key = `${action.payload.locale}:${action.payload.path}`;
return {
  ...state,
  [key]: {
    ...(state[key]),
    isFetching: true,
  },
}
```
`key` should be extracted to standalone function based on `action.payload`

File: src/reducers/content.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    case GET_LOCALE_CONTENT_SUCCESS:
+      return {
+        ...state,
+        isFetching: false,
+        data: action.payload.data,
+      };
```
Comment (langpavel) [inline src/reducers/content.js]:
```js
return {
  ...state,
  [key]: {
    isFetching: false,
    ...action.payload
  },
};
```

File: src/routes/content/Content.js [hunk 6]
```diff
@@ -7,28 +7,54 @@
...
 }
 
-Content.propTypes = {
-  path: PropTypes.string.isRequired,
-  content: PropTypes.string.isRequired,
-  title: PropTypes.string,
+const mapState = (state) => ({
+  locale: state.intl.locale,
+  data: state.content.data,
+});
```
Comment (langpavel) [inline src/routes/content/Content.js]:
```js
const mapState = (state, props) => {
  const locale = state.intl.locale;
  const content = state.content[`${locale}:${props.path}`];
  return {
    locale: state.intl.locale,
    ...content
  };
};
```

**Comment by awesomejerry** [comment]
Thanks for the code reviews and suggestions! They're quite helpful.

**Comment by langpavel** [comment]
Thanks @awesomejerry 
I refactored some stuff to agressively cache content on client. Have a look!

**Pull request merged**